src/request.rs

```diff
@@ -65,25 +41,38 @@ impl fmt::Debug for Request {
         write!(
             f,
             "Request({} {}, {:?})",
-            self.method, self.parsed_url, self.headers
+            self.method, self.url, self.headers
         )
     }
 }
 
 impl Request {
     pub(crate) fn new(agent: Agent, method: String, url: String) -> Request {
-        Self::_new(agent, method, url.into())
+        Self::_new(agent, method, url)
     }
 
     pub(crate) fn with_url(agent: Agent, method: String, url: Url) -> Request {
-        Self::_new(agent, method, url.into())
+        Self::_new(agent, method, url.to_string())
     }
 
-    fn _new(agent: Agent, method: String, parsed_url: ParsedUrl) -> Request {
+    fn _new(agent: Agent, method: String, url: String) -> Request {
+        let query_pairs = if let Ok(url) = url.parse::<Url>() {
+            let mut vec = vec![];
+
+            for (k, v) in url.query_pairs() {
+                vec.push((k.into(), v.into()));
+            }
+
+            vec
+        } else {
+            vec![]
+        };
```

Comment:
I think this can be simplified like:

```
let query_pairs: Vec<(String, String)> = match url.parse::<Url>() {
  Ok(u) => u.query_pairs().collect(),
  Err(e) => vec![],
}
```

Reply:
done